runtime/compare.c

```diff
@@ -288,8 +280,8 @@ static intnat do_compare_val(struct compare_stack* stk,
       if (sz1 > 1) {
         sp++;
         if (sp >= stk->limit) sp = compare_resize_stack(stk, sp);
-        sp->v1 = &Field(v1, 1);
-        sp->v2 = &Field(v2, 1);
+        sp->v1 = Op_val(v1) + 1;
```

Comment:
Consider reverting for smaller diff.

Reply:
Fixed in https://github.com/ocaml-multicore/ocaml/commit/5eb99707fcfe22153a0fbb33c9bd508f75375be3